src/Utility/Vec3.h

```diff
@@ -216,6 +219,9 @@ class Vec4 : public Vec3
   { x = X.x; y = X.y; z = X.z; t = X.t; idx = X.idx; u = X.u; return *this; }
   //! \brief Overloaded assignment operator.
   Vec4& operator=(Real val) { x = y = z = val; return *this; }
+  //! \brief Overloaded assignment operator.
+  Vec4& operator=(const std::vector<Real>& X)
+  { this->Vec3::operator=(X); return *this; }
```

Comment:
If we need these overloads, shouldn't it also be a `t = X.size() > 3 ? X[3] : 0.0; idx = -1; ` in the latter to be consistent with the `Vec4(const std::vector<Real>& X)` constructor?

Alternatively, one could use `Vec4::assign(const Vec3&)` instead of this assign operator, which does not blank out the `u` pointer when the argument is not a `Vec4`.

Reply:
then i need to change all the X = Xnod * N; in integration methods.